EIPS/eip-3000.md

```diff
@@ -0,0 +1,146 @@
+
+    /**
+     * @notice Executes an action after its execution delayed has passed and its state hasn't been altered by a challenge or veto
+     * @param container A ERC3000Data.Container struct holding both the paylaod being scheduled for execution and
+       the current configuration of the system
+     * should be a MUST payload.executor.exec(payload.actions)
+     */
+    function execute(ERC3000Data.Container memory container) virtual public returns (bytes[] memory execResults);
+    event Executed(bytes32 indexed containerHash, address indexed actor, bytes[] execResults);
+
+    /**
+     * @notice Challenge a container in case its scheduling is illegal as per Config.rules. Pulls collateral and dispute fees from sender into contract
+     * @param container A ERC3000Data.Container struct holding both the paylaod being scheduled for execution and
+       the current configuration of the system
+     * @param reason Hint for case reviewers as to why the scheduled container is illegal
+     */
+    function challenge(ERC3000Data.Container memory container, bytes memory reason) virtual public returns (uint256 resolverId);
+    event Challenged(bytes32 indexed containerHash, address indexed actor, bytes reason, uint256 resolverId, ERC3000Data.Collateral collateral);
+
+    /**
+     * @notice Apply arbitrator's ruling over a challenge once it has come to a final ruling
+     * @param container A ERC3000Data.Container struct holding both the paylaod being scheduled for execution and
+       the current configuration of the system
+     * @param resolverId disputeId in the arbitrator in which the dispute over the container was created
+     */
+    function resolve(ERC3000Data.Container memory container, uint256 resolverId) virtual public returns (bytes[] memory execResults);
+    event Resolved(bytes32 indexed containerHash, address indexed actor, bool approved);
+
+    /**
+     * @dev OPTIONAL
+     * @notice Apply arbitrator's ruling over a challenge once it has come to a final ruling
+     * @param payloadHash Hash of the payload being vetoed
+     * @param config A ERC3000Data.Config struct holding the config attached to the payload being vetod
+     */
+    function veto(bytes32 payloadHash, ERC3000Data.Config memory config, bytes memory reason) virtual public;
+    event Vetoed(bytes32 indexed containerHash, address indexed actor, bytes reason, ERC3000Data.Collateral collateral);
+
+    /**
+     * @dev OPTIONAL: implementer might choose not to implement (initial Configured event MUST be emitted)
+     * @notice Apply a new configuration for all *new* containers to be scheduled
+     * @param config A ERC3000Data.Config struct holding all the new params that will control the queue
+     */
+    function configure(ERC3000Data.Config memory config) virtual public returns (bytes32 configHash);
+    event Configured(bytes32 indexed containerHash, address indexed actor, ERC3000Data.Config config);
+}
+```
+
+## Implementations
```

Comment:
```suggestion
## Rationale
TBD

## Implementations
```
Rationale is a required section.  It should focus on stating the reason why specific decisions were made over alternate choices.  Some of the content currently is the Motivation would be more appropriate in the Rationale.